src/eos-updater-poll-common.c

```diff
@@ -374,14 +374,13 @@ get_extensions_url (OstreeRepo *repo,
                     gchar **extensions_url,
                     GError **error)
 {
-  g_autofree gchar *url = NULL;
+  g_autofree gchar *url = g_strdup (url_override);
 
-  if (url_override != NULL)
-    url = g_strdup (url_override);
-  else if (!ostree_repo_remote_get_url (repo, remote_name, &url, error))
+  if (url == NULL &&
+      !ostree_repo_remote_get_url (repo, remote_name, &url, error))
     return FALSE;
 
-  *extensions_url = g_strconcat (url, "/extensions/eos", NULL);
+  *extensions_url = g_build_path ("/", url, "extensions", "eos", NULL);
```

Comment:
It is `g_build_path`, not `g_build_filename`. The former is more generic, so it takes a delimiter as a first parameter.